WalkAndTravel/ClassLibrary/Levels.cs

```diff
@@ -49,5 +49,15 @@ public void CreateTrail(int exp)
         {
             experience += exp;
         }
+
+        public int Narrowing(double exp)
+        {
+            return (int)exp;
+        }
+
+        public double Widening(int exp)
+        {
+            return (double)exp;
+        }
```

Comment:
Maybe change these two functions into one that calculates how much experience must be given after completing some length route and there you will be able to use these narrowing and widening type conversions.